### PR TITLE
Update Grafana to version 11 and provision dashboards from usegalaxy-eu/grafana-dashboards

### PR DESCRIPTION
**Pull request by kysrpex**
Update Grafana to version 11 and switch from deprecated role `cloudalchemy.grafana` to the official Grafana role from the `grafana.grafana` collection. Update `usegalaxy_eu.grafana_matrix_forwarder`. Instead of disabling firewalld on the Grafana host, open the nginx ports.

Before merging this PR, the Grafana database must be migrated from SQLite to PostgresSQL. This is rather simple using [pgloader](https://pgloader.readthedocs.io/en/latest/index.html) and following [this guide](https://polyglot.jamie.ly/programming/2019/07/01/grafana-sqlite-to-postgres) from [Jamie Ly](https://resume.jamie.ly/). The postgres database information can be found [here](https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1235/files#diff-c967ddc3e213dc80d7efac982b95ffcaf62ead091534366282be6540354708d3R53-R58). It can be accessed from stats, sn06 and maintenance (see [usegalaxy-eu/infrastructure-playbook#1235](https://github.com/usegalaxy-eu/infrastructure-playbook)).

Migration in a nutshell:

1. Make sure the `grafana` database exists in Postgres and the `grafana` user has all privileges on it. You may run [reset.sql](https://github.com/user-attachments/files/16234900/reset.sql.zip) as `postgres` to do this ( :warning: this drops the `grafana` database).
2. Run [schema.sql](https://github.com/user-attachments/files/16234899/schema.sql.zip) as `grafana` to create all tables Grafana v9.2.10 (c37dcaf0da) needs. Grafana v11 will run database migrations on top of it.
3. Install [pg_loader](https://pgloader.readthedocs.io/en/latest/index.html). Luckily, [there are packages available in the Postgres community repository](https://pgloader.readthedocs.io/en/latest/install.html#rpm-packages). To enable the repo, you just need the right [repository package](https://yum.postgresql.org/repopackages/). You may (and should) delete the repository package once you have completed this migration.
4. Open [grafana.load](https://github.com/user-attachments/files/16234901/grafana.load.zip) and adjust the path to `/data/monitoring/grafana_data/grafana.db` (if running the migration steps on stats.galaxyproject.eu) and the [PostgreSQL connection string](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) if needed.
5. Run `pgloader grafana.load`. Warnings and even errors are expected, not an issue.
6. Merge this PR and run the grafana.yml playbook.
7. Enable the [stats-grafana Jenkins project](https://build.galaxyproject.eu/job/usegalaxy-eu/job/playbooks/job/stats-grafana/).
8. Some dashboards (the provisioned ones) will be in the wrong folder. Move them to the correct folder (see pictures below). All the remaining dashboards that do not belong to an existing folder should be moved to a new folder "General".

Bildschirmfoto vom 2024-07-15 15-21-52
Bildschirmfoto vom 2024-07-15 15-22-19
Bildschirmfoto vom 2024-07-15 15-22-36

Closes usegalaxy-eu/issues#558.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "mounts"]
 	path = mounts
 	url = https://github.com/usegalaxy-eu/mounts
+[submodule "files/grafana"]
+	path = files/grafana
+	url = https://github.com/usegalaxy-eu/grafana-dashboards.git
```
Comment (mira-miracoli) [inline .gitmodules:6]: I am not sure if the `.git` could lead to problems, but I am also not an expert, was just noticing, that mount doesn't have it

Comment (kysrpex) [reply .gitmodules:6]: Should be fine afaik, that's even what GitHub suggests

grafik

Comment (mira-miracoli) [reply .gitmodules:6]: ok :)
I just saw it at [git handbook](https://git-scm.com/book/en/v2/Git-Tools-Submodules) without and was not sure about this

File: group_vars/grafana/vars.yml
[the remark this answers is not in the document]
Comment (kysrpex) [reply group_vars/grafana/vars.yml:1]: As fellow human being, I also push personal preferences :laughing: Some advantages of this schema are that you [do not have](https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1235/files#diff-8d0c3127a139fe96af7ed65a80c62a7cddd117794db6b15a9680318a7a587a04R4-R5) to [declare vars files](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/92ddb8832c83e6986b8b285b2de6c698e0ac979d/grafana.yml#L9), and other playbooks do not load secrets they do not need to know.

**Review by mira-miracoli — APPROVED** [review]
Thank you!

**Comment by bgruening** [comment]
Cool beans! Thanks a lot!

I have a technical question, but please deploy.
Old links to dashboards will probably not work anymore, I assume, is there a way in grafana to keep stable links? Or should we use bitly or even better gx.io to create stable links dashboards?

**Comment by kysrpex** [comment]
> Cool beans! Thanks a lot!
> 
> I have a technical question, but please deploy. Old links to dashboards will probably not work anymore, I assume, is there a way in grafana to keep stable links? Or should we use bitly or even better gx.io to create stable links dashboards?

With this migration path all dashboards keep the same uids, and Grafana has not changed how it forms the urls, so links do not break (see screenshots below).

Bildschirmfoto vom 2024-07-16 09-01-57
Bildschirmfoto vom 2024-07-16 09-02-13

**Comment by bgruening** [comment]
Oh nice!!!

**Comment by bgruening** [comment]
Can you please also sum that all up in a nice blog post and in addition in a operations update.

Thanks a lot!

**Review by mira-miracoli — APPROVED** [review]

**Comment by mira-miracoli** [comment]
oh I guess my github was stuck

**Comment by kysrpex** [comment]
@sanjaysrikakulam @mira-miracoli Important detail. You may find that some panels do not work. You can very easily fix them like this:

1. Find a panel that does not work.
Bildschirmfoto vom 2024-07-23 15-59-20

2. Click "Edit".
Bildschirmfoto vom 2024-07-23 15-59-34

3. Click the pencil on the bottom right (raw query mode). The preview will start working instantly.
Bildschirmfoto vom 2024-07-23 16-00-01
Bildschirmfoto vom 2024-07-23 16-00-20

4. Save the dashboard.
Bildschirmfoto vom 2024-07-23 16-00-34

Don't ask me why this works. Actually if you switch the panel back to visual editor mode it breaks again.

**Comment by bgruening** [comment]
This one https://stats.galaxyproject.eu/d/000000004/galaxy?orgId=1&refresh=10s&viewPanel=7 does not seem to work. Maybe wrong host?

**Comment by kysrpex** [comment]
> This one https://stats.galaxyproject.eu/d/000000004/galaxy?orgId=1&refresh=10s&viewPanel=7 does not seem to work. Maybe wrong host?

The list of hosts is populated by the InfluxQL query `SHOW TAG VALUES FROM "cluster.queue" WITH KEY = "host"`, but the only possible outcome of that query right now is "maintenance.galaxyproject.eu", see the InfluxDB measurement query below.

```
> SELECT * FROM "cluster.queue" ORDER BY time desc LIMIT 10
name: cluster.queue
time                 count engine host                         schedd                state
----                 ----- ------ ----                         ------                -----
2024-07-24T12:38:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu suspended
2024-07-24T12:38:00Z 539   condor maintenance.galaxyproject.eu sn06.galaxyproject.eu running
2024-07-24T12:38:00Z 426   condor maintenance.galaxyproject.eu sn06.galaxyproject.eu idle
2024-07-24T12:38:00Z 2     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu held
2024-07-24T12:38:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu removed
2024-07-24T12:38:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu completed
2024-07-24T12:37:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu removed
2024-07-24T12:37:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu completed
2024-07-24T12:37:00Z 0     condor maintenance.galaxyproject.eu sn06.galaxyproject.eu suspended
2024-07-24T12:37:00Z 543   condor maintenance.galaxyproject.eu sn06.galaxyproject.eu running
```
Changing to `SHOW TAG VALUES FROM "cluster.queue" WITH KEY = "schedd"` allows choosing `sn06.galaxyproject.eu` too, and then the route timings work.

This more likely has to do with the setup of the maintenance node and the HTCondor migration than with the Grafana update.

Note to all: if you see fixable problems like these on the dashboards go ahead and fix them.

**Pull request merged**